Bindings/Packages/com.virtualmaker.bindings/Runtime/Bindings.PlayerPrefs.cs

```diff
@@ -0,0 +1,97 @@
+using UnityEngine;
+
+namespace VirtualMaker.Bindings
+{
+    public partial class Bindings
+    {
+        /// <summary> Binds player pref to property. </summary>
+        /// <param name="key"> Key used by <c>PlayerPrefs</c> to get and set the value. </param>
+        /// <param name="prop"> Property that sets the value. </param>
+        /// <param name="getFromPref"> Whether to try setting <c>prop.Value</c> to the value already inside the <c>PlayerPrefs</c>. </param>
+        /// <param name="defaultValue"> Value to set <c>prop.Value</c> to if <c>getFromPref</c> is <c>true</c> but there isn't a value already inside the <c>PlayerPref</c>. </param>
+        public void BindPlayerPref(string key, Property<float> prop, bool getFromPref, float? defaultValue = null)
+        {
+            if (getFromPref)
+            {
+                prop.Value = PlayerPrefs.GetFloat(key, defaultValue ?? prop.Value);
+            }
+            BindPlayerPref(key, prop);
+        }
+
+        /// <summary> Binds player pref to property. </summary>
+        /// <param name="key"> Key used by <c>PlayerPrefs</c> to set the value. </param>
+        /// <param name="prop"> <c>IProperty</c> that sets the value. </param>
+        public void BindPlayerPref(string key, IProperty<float> prop)
+        {
+            Bind(prop, value => PlayerPrefs.SetFloat(key, value));
+            SavePrefsOnReset();
+        }
+
+        /// <summary> Binds player pref to property. </summary>
+        /// <param name="key"> Key used by <c>PlayerPrefs</c> to get and set the value. </param>
+        /// <param name="prop"> Property that sets the value. </param>
+        /// <param name="getFromPref"> Whether to try setting <c>prop.Value</c> to the value already inside the <c>PlayerPrefs</c>. </param>
+        /// <param name="defaultValue"> Value to set <c>prop.Value</c> to if <c>getFromPref</c> is <c>true</c> but there isn't a value already inside the <c>PlayerPref</c>. </param>
+        public void BindPlayerPref(string key, Property<int> prop, bool getFromPref, int? defaultValue = null)
+        {
+            if (getFromPref)
+            {
+                prop.Value = PlayerPrefs.GetInt(key, defaultValue ?? prop.Value);
```

Comment:
Not sure if we want to set prop.Value as default if no default value passed

Reply:
Maybe just remove getFromPref and have this version always take a default value. Why would you pass in a `Property` instead of an `IProperty` if you don't want to get the value.

If the overloads are confusing, just rename this one to something like `GetAndBindPlayerPref`

Reply:
or just have the caller write two lines of code :D

Reply:
> If the overloads are confusing, just rename this one to something like `GetAndBindPlayerPref`
The overloads were confusing me, yes. I was trying to redo the `bool twoWay` in `BindField`, but yes the extra default value param risks being confusing.

> or just have the caller write two lines of code :D
If all these methods do is call one super simple line, then I wouldn't see the point in adding these. If it won't reduce the line count for the person using this, just keep to the lambdas.

Reply:
I don't feel great about introducing a `GetAndBind` when that hasn't been the case for the likes of `BindToggle`, `BindSlider`, nor `BindField`.

Maybe to make them match better I get rid of the `IProperty` versions for `BindPlayerPref` that the `twoWay` binding methods don't have?

Reply:
Made changes to it; I think the summary will help avoid the overload confusions. Resolving this conversation so if we discuss it further then it'll be looking over the up-to-date rendition.